Functions/Solution.cs

```diff
@@ -15,74 +16,267 @@ internal class Solution
+                {"Paper", RockPaperScissors.RPS.Paper},
+                {"Scissors", RockPaperScissors.RPS.Scissors},
+            };
+            
+            try
+            {
+                Console.WriteLine("Enter player 1 move.");
+                RockPaperScissors.RPS move1 = strToMove[Console.ReadLine()];
+                Console.WriteLine("Enter player 2 move.");
+                RockPaperScissors.RPS move2 = strToMove[Console.ReadLine()];
+
+                if (move1 == move2)
+                    Console.WriteLine("Tie.");
+                else if ((move1 == RockPaperScissors.RPS.Rock && move2 == RockPaperScissors.RPS.Paper)
+                        || move1 == RockPaperScissors.RPS.Paper && move2 == RockPaperScissors.RPS.Scissors
+                        || move1 == RockPaperScissors.RPS.Scissors && move2 == RockPaperScissors.RPS.Rock
+                        )
+                    Console.WriteLine("Player 2 wins.");
+                else
+                    Console.WriteLine("Player 1 wins.");
+            }
+            catch (System.Collections.Generic.KeyNotFoundException)
+            {
+                Console.WriteLine("Not a legal move.");
+            }
         }
 
         public static double[] Assignment4(bool asc, params double[] nums)
         {
-            /*
-             * TODO: Write code here
-             */
+            int n = nums.Length;
+            double[] arr = new double[n];
+            
+            nums.CopyTo(arr, 0);
+
+            double tmp;
+            for(int i=0; i<n; i++)
+            {
+                for(int j=i+1; j<n; j++)
+                {
+                    if(arr[j] < arr[i])
+                    {
+                        tmp = arr[i];
+                        arr[i] = arr[j];
+                        arr[j] = tmp;
+                    }
+                }
+            }
+
+            // change sort direction if needed
+            if (asc == false)
+            {
+                double[] final = new double[n];
+                for(int i=0; i<n; i++)
+                    final[i] = arr[n-1-i];
+                return final;
+            }
+            else
+                return arr;
         }
 
         public static void Assignment5(string str, char[] letters)
         {
-            /*
-             * TODO: Write code here
-             */
+            const int letterOffset = 10;
+            var newStr = new StringBuilder();
+            foreach (char c in str)
+            {
+                bool charInLetters = false;
+                Console.WriteLine();
+                foreach (char letter in letters)
+                {
+                    if (c == letter)
+                    {
+                        charInLetters = true;
+                        if (c + letterOffset > 'z')
+                            {
+                                newStr.Append((char)('a' + 'z' - c));
+                                break;
```

Comment:
You can break either way once you know that `c == letter`.

Reply:
Nice spotting.